src/app/utils/socket-utils.ts

```diff
@@ -175,6 +169,21 @@ export function plugFitsIntoSocket(socket: DimSocket, plugHash: number) {
   );
 }
 
+/**
+ * For a "choice socket" (a socket where all options are equivalent and
+ * there's no empty option), like Subclass supers or abilities, or the
+ * Aeon cult plugs, find whatever should be equipped as a default.
+ */
```

Comment:
this ends up non-null asserted several times. maybe a small in-jsdoc note about when it's safe to non-null assert this?

or a note that it should only be run at all against a certain type of socket, and `as` the non-null-ness into the return type itself